modules/serverpod_auth/serverpod_auth_server/lib/src/endpoints/status_endpoint.dart

```diff
@@ -16,9 +16,17 @@ class StatusEndpoint extends Endpoint {
     return userId != null;
   }
 
-  /// Signs out a user.
-  Future<void> signOut(Session session) async {
-    await UserAuthentication.signOutUser(session);
+  /// Signs out a user from the server. If [allSessions] is true, the user is signed out
+  /// from all active sessions by deleting all authentication keys. If [allSessions]
+  /// is false, only the current session's authentication key is deleted.
+  Future<void> signOut(
+    Session session, {
+    required bool allSessions,
```

Comment:
Breaking: How will this affect already existing clients? If I understand this change correctly, all requests from old clients will fail now since this is a required parameter in the endpoint.

In the specification of the issue this was listed as additional requirements:

> This change needs to be done in a backwards-compatible way in the network layer so that old clients are still supported with preserved behavior.
> 
> This could be a configurable option so that developers can choose to migrate old clients over to signing out only a single device.

Please ensure this is respected in the change.

Reply:
@SandPod The issue has been addressed by refactoring the code (the code above) and replacing it with new methods. While some parts of the original code may still be present, it has largely been updated. The changes also ensure backward compatibility, so old clients should still function as expected without requiring immediate changes on their end.